core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -1150,15 +1147,16 @@ private void copyLibraryJars(FileSystem runnerZipFs, OutputTargetBuildItem outpu
                     if (transformedFromThisArchive == null || transformedFromThisArchive.isEmpty()) {
                         final String fileName = appDep.getGroupId() + "." + resolvedDep.getFileName();
                         final Path targetPath = libDir.resolve(fileName);
-                        Files.copy(resolvedDep, targetPath, StandardCopyOption.REPLACE_EXISTING);
+                        // Unsign the jar before copying it
+                        JarUnsigner.unsignJar(resolvedDep, targetPath);
```

Comment:
@maxandersen this is the fix, it unsigns the JAR before copying it to the lib/ dir